### DISCUSSION TITLE
Restrict User

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#1143] [Help] [answered]
Is there any way to stop groups of users from accessing a Resource altogether? 

I have bunch of groups - one of them is called **Observer** group. They can basically view the data etc., which I have set up using Policy in my models. But I don't want to at all allow them any access to some of the Resources. For example, I have a navigation group called **User Management** block, in which there are *UserResource*, *RoleResource* and a couple of more things about configuration. I don't want them to view any of these resources. Is there a way to accomplish it? Something like "hidden" or "visible" (as is used for forms) but for the entire resource?

**Comment by danharrin** [comment, marked as answer]
Hey, the `viewAny()` policy method is used to completely hide the resource. https://filamentadmin.com/docs/2.x/admin/resources#authorization

**Reply by anishsarkar** [reply]
Oh. Great! Thanks a ton.

**Reply by khanakia** [reply]
@danharrin Update Policy does not enforce when i copy the request as CURL and use Postman to send the request and can update any record. 

e.g.

I have this policy
```
public function update(User $user, UserPaymentMethod $userPaymentMethod)
    {
        return $user->id==$userPaymentMethod->user_id;
    }
```

and if send this request using postman as any user it simply updates the record.
```
{
    "fingerprint": {
        "id": "DZUzjoPAxTOQTisCGkVd",
        "name": "app.filament.resources.user-payment-method-resource.pages.edit-user-payment-method",
        "locale": "en",
        "path": "admin/user-payment-methods/1/edit",
        "method": "GET",
        "v": "acj"
    },
    "serverMemo": {
        "children": [],
        "errors": [],
        "htmlHash": "6885ea83",
        "data": {
            "data": {
                "id": 1,
                "created_at": "2022-08-06T06:59:00.000000Z",
                "updated_at": "2022-08-06T07:09:16.000000Z",
                "deleted_at": null,
                "company_id": 1,
                "name": "jkh11",
                "status": false,
                "acc_no": "jkh",
                "acc_name": "kjh",
                "bank_name": "jkh",
                "ifsc": "jkh",
                "acctype": "hj",
                "user_id": null
            },
            "mountedAction": null,
            "mountedActionData": [],
            "componentFileAttachments": [],
            "mountedFormComponentAction": null,
            "mountedFormComponentActionData": [],
            "mountedFormComponentActionComponent": null,
            "activeRelationManager": null,
            "record": []
        },
        "dataMeta": {
            "models": {
                "record": {
                    "class": "App\\Models\\UserPaymentMethod",
                    "id": 1,
                    "relations": [],
                    "connection": "mysql"
                }
            }
        },
        "checksum": "ef207c96a87758827851bfd9187855fad768c2cc8a3321a5ad6b81a6e1967b0e"
    },
    "updates": [
        {
            "type": "callMethod",
            "payload": {
                "id": "ii3o",
                "method": "save",
                "params": []
            }
        }
    ]
}
```

**Reply by danharrin** [reply]
It is enforced. It's enforced when the Livewire component is mounted, and if that doesnt happen, then you are unable to get the checksum (`ef207c96a87758827851bfd9187855fad768c2cc8a3321a5ad6b81a6e1967b0e`) or ID (`DZUzjoPAxTOQTisCGkVd`) of the Livewire component. Without those, you cannot make a successful request.

**Reply by khanakia** [reply]
I see so then nothing to worry about i believe 👍

**Reply by danharrin** [reply]
Yeah, try removing or changing the checksum, the request will not work :)
